govcd/vm_test.go

```diff
@@ -1963,16 +1698,34 @@ func (vcd *TestVCD) Test_CreateStandaloneVM(check *C) {
 				Info:         "Specifies Guest OS Customization Settings",
 				ComputerName: "standalone1",
 			},
+			BootOptions: &types.BootOptions{
+				BootDelay:      addrOf(0),
+				EnterBiosSetup: addrOf(true),
+			},
 		},
 		Xmlns: types.XMLNamespaceVCloud,
 	}
+
+	if vcd.client.Client.APIVCDMaxVersionIs(">=37.1") {
+		params.CreateVm.VmSpecSection.Firmware = "efi"
+		params.CreateVm.BootOptions.EfiSecureBootEnabled = addrOf(true)
+		params.CreateVm.BootOptions.BootRetryEnabled = addrOf(true)
+		params.CreateVm.BootOptions.BootRetryDelay = addrOf(20)
+	}
+
 	vappList := vdc.GetVappList()
 	vappNum := len(vappList)
-	vm, err := vdc.CreateStandaloneVm(&params)
+	vm, err := vdc.CreateStandaloneVm(params)
 	check.Assert(err, IsNil)
 	check.Assert(vm, NotNil)
+	err = vm.Refresh()
+	check.Assert(err, IsNil)
 	AddToCleanupList(vm.VM.ID, "standaloneVm", "", check.TestName())
 	check.Assert(vm.VM.Description, Equals, description)
+	check.Assert(vm.VM.BootOptions.EnterBiosSetup, DeepEquals, addrOf(true))
+	check.Assert(vm.VM.BootOptions.EfiSecureBootEnabled, DeepEquals, addrOf(true))
+	check.Assert(vm.VM.BootOptions.BootRetryEnabled, DeepEquals, addrOf(true))
+	check.Assert(vm.VM.BootOptions.BootRetryDelay, DeepEquals, addrOf(20))
```

Comment:
Will this work on API versions < 37.1? (10.4.0) given that just above some of the settings are being set only in API 37.1?